cli/tests/053_share_resources.ts

```diff
@@ -0,0 +1,40 @@
+interface Deferred<T> extends Promise<T> {
+  resolve: (value?: T | PromiseLike<T>) => void;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  reject: (reason?: any) => void;
+}
+function deferred<T>(): Deferred<T> {
+  let methods;
+  const promise = new Promise<T>((resolve, reject): void => {
+    methods = { resolve, reject };
+  });
+  return Object.assign(promise, methods)! as Deferred<T>;
+}
+
+interface TaggedYieldedValue<T> {
+  iterator: AsyncIterableIterator<T>;
+  value: T;
+}
+
+const _f = Deno.openSync("./shared_resource_worker.ts");
+console.log(Deno.resources());
+
+const w0 = new Worker("./shared_resource_worker.ts", {
+  shareResources: true
+});
+const w0Deferred = deferred();
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+w0.onmessage = (_e: any): void => {
+  w0Deferred.resolve();
+};
+await w0Deferred;
+
+const w1 = new Worker("./shared_resource_worker.ts");
+const w1Deferred = deferred();
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+w1.onmessage = (_e: any): void => {
+  w1Deferred.resolve();
+};
+await w1Deferred;
+
+Deno.exit(0);
```

Comment:
Is this necessary? Can you add a comment describing why this needs to be done?

Reply:
There is a small problem in our worker design: currently we require calling `workerClose()` to close a worker. However, it turns out due to structure of `workerMain()`, it is only effective when `workerClose()` is called inside `onmessage`, which I don't have on the workers here.

I'll work around with this limitation using `onmessage` in this PR since they are not necessarily related, and that Bartek mentioned about `Worker` API conforming refactoring (that we are likely be get rid of `workerClose` soon)